starlight.sh

```diff
@@ -14,22 +14,38 @@ cmake $SOURCEDIR -DCMAKE_INSTALL_PREFIX=$INSTALLROOT       \
                  -DCMAKE_BUILD_TYPE=$CMAKE_BUILD_TYPE      \
                  -DCMAKE_SKIP_RPATH=TRUE                   \
                  -DENABLE_HEPMC3=ON                        \
+		 -DCMAKE_EXPORT_COMPILE_COMMANDS=ON        \
+		 -DBUILD_SHARED_LIB=ON                     \
                  -DHepMC3_DIR="$HEPMC3_ROOT"
-cmake --build . -- ${JOBS:+-j$JOBS}
-mkdir -p $INSTALLROOT/bin
-cp ./starlight $INSTALLROOT/bin/.
-mkdir -p $INSTALLROOT/lib
-cp ./libStarlib.a $INSTALLROOT/lib/.
+
+cmake --build . -- ${JOBS:+-j$JOBS} install
 cp -r $SOURCEDIR/config $INSTALLROOT/.
 cp -r $SOURCEDIR/include $INSTALLROOT/.
 
+# We need to fix the installation of STARlight, in particular
+# header files. They define global macros which clash with other code when
+# included in ROOT macros.
+sed -i 's/printInfo/starlight_printInfo/' ${INSTALLROOT}/include/*.h
+sed -i 's/printWarn/starlight_printWarn/' ${INSTALLROOT}/include/*.h
+sed -i 's/printErr/starlight_printErr/' ${INSTALLROOT}/include/*.h
+
 #ConfigFile
 cat << EOF > $INSTALLROOT/bin/starlight-config
 echo $INSTALLROOT
 EOF
 chmod +x $INSTALLROOT/bin/starlight-config
 
-#ModuleFile
-mkdir -p etc/modulefiles
-alibuild-generate-module --bin --lib > etc/modulefiles/$PKGNAME
-mkdir -p $INSTALLROOT/etc/modulefiles && rsync -a --delete etc/modulefiles/ $INSTALLROOT/etc/modulefiles
+# Modulefile
+MODULEDIR="$INSTALLROOT/etc/modulefiles"
+MODULEFILE="$MODULEDIR/$PKGNAME"
+mkdir -p "$MODULEDIR"
+alibuild-generate-module --bin --lib > $MODULEFILE
```

Comment:
Given STARLight is CMake based, I think it would be better to provide `--cmake` to `alibuild-generate-module` and avoid the custom `$STARLIGHT_ROOT` variable.